pkg/log/logger.go

```diff
@@ -9,9 +9,13 @@ import (
 )
 
 func ConfigLogger(level int8, syncer zapcore.WriteSyncer) logr.Logger {
-	if level > 0 {
-		level = -level
```

Comment:
@Tomasz-Smelcerz-SAP 

I still don't fully get it, but seems that this was right. According to docs, `zap` is using log levels like Debug -1, Info 0, Warning 2, ... where higher level means higher importance and less verbosity.

We are logging through the `logr` API where `zapr` is the implementation using `zap`. `logr` does not have a notion of log levels. Instead, it just distinguishes `info` and `error` messages and assigns a verbosity. The verbosity means "it is a more or less important info log" for instance. Verbosities are positive where higher number means less importance (opposite of `zap`).

Now, `zapr` describes the relation here https://github.com/go-logr/zapr?tab=readme-ov-file#increasing-verbosity where they initialize with the negative levels. Here I am now a bit confused and still need to think about it further. Just wanted to share my thoughts already.

Reply:
There we go: https://pkg.go.dev/github.com/go-logr/zapr#hdr-Implementation_Details

*Levels in logr correspond to custom debug levels in Zap. Any given level in logr is represents by its inverse in zap (`zapLevel = -1*logrLevel`). For example V(2) is equivalent to log level -2 in Zap, while V(1) is equivalent to Zap's DebugLevel.*